File: src/main/java/com/graqr/threshr/model/redsky/store/Day.java
package com.graqr.threshr.model.redsky.store;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record Day(
        @JsonProperty("is_open") Boolean isOpen,
        String date,
        @JsonProperty("day_name")
        String dayName,
        List<Hour> hours,
        @Nullable @JsonProperty("sequence_number")
        String sequenceNumber) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/store/Day.kt
package com.graqr.threshr.model.redsky.store

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Day(
    @param:JsonProperty("is_open") val isOpen: Boolean,
    val date: String,
    @param:JsonProperty("day_name")
    val dayName: String,
    val hours: List<Hour>,
    @param:Nullable @param:JsonProperty("sequence_number")
    val sequenceNumber: String
) {
}